src/cmd.rs

```diff
@@ -480,6 +486,9 @@ impl Pipeline {
     ///     .cmd("GET").arg("key_1")
     ///     .cmd("GET").arg("key_2").query(&con).unwrap();
     /// ```
+    ///
+    /// NOTE: In order to clear (and reuse) the `Pipeline` object after calling `query()`, one must
```

Comment:
I think the `reuse` here is slightly misleading.
Without clear the pipeline can be reused with the existing commands in it, `clear` gives you a clean pipeline without losing the allocation.

Maybe something like:

> NOTE: In order to clear and reuse an empty `Pipeline` object ... 

What do you think?

Reply:
What about:

> NOTE: A `Pipeline` object may be reused after `query()` with all the commands as were inserted to them. In order to clear a `Pipeline` object with minimal memory released/allocated, one should call the `clear()`method before inserting new commands.

Reply:
Probably "function" is better than "method".